### ISSUE TITLE
Support For Creating an Answer with Passive Negotiation

### ISSUE THREAD

**Issue by Step7750** [pion/webrtc#494]
RFC 5763 - Section 5
```
The endpoint MUST use the setup attribute defined in [RFC4145].
The endpoint that is the offerer MUST use the setup attribute
value of setup:actpass and be prepared to receive a client_hello
before it receives the answer.  The answerer MUST use either a
setup attribute value of setup:active or setup:passive.  Note that
if the answerer uses setup:passive, then the DTLS handshake will
not begin until the answerer is received, which adds additional
latency. setup:active allows the answer and the DTLS handshake to
occur in parallel.  Thus, setup:active is RECOMMENDED.  Whichever
party is active MUST initiate a DTLS handshake by sending a
ClientHello over each flow (host/port quartet).
```

Currently, pions/webrtc assumes that if the remote SDP is an `RTCSdpTypeOffer` that we must reply with `setup:active` and initiate the DLTS handshake. There should be an option to be the passive party in the connection if we're creating an answer.

I could probably make a PR for this if need be.

**Comment by backkem** [comment]
Yes, you are right. We still make this assumption.

We'd love to have you if you want to take this one on. Feel free to ask any questions you may have. We're also have an active community in the #pion channel of the Golang Slack.

**Comment by notedit** [comment]
@Step7750 I came to this also.

**Comment by Sean-Der** [comment]
@notedit @Step7750 I pushed a PR that should fix this. I added tests locally, but haven't been able to confirm this against a real application.

How were you able to reproduce this? Anything I can easily test against? thanks!

**Comment by notedit** [comment]
@Sean-Der Thanks,  I will test this next week.

**Comment by notedit** [comment]
It works in my local test.

**Comment by Sean-Der** [comment]
Fantastic! will merge into master, thanks for testing @notedit

**Comment by jbert** [comment]
Hi. In our testing, we see that setup:passive is now the default if the settingengine is not used.
i.e. we see:

offer (setup:actpass)
pion/webrtc: answer (setup:passive)

This is permitted by the RFC, but not recommended due to potentially added latency.
I haven't tried to quantify any additional latency, but this change does break interop with a latency-sensitive client (using the settingengine to change the default does change this).

Do you agree we should change the default?

Would a PR to change the defaultDtlsRoleAnswer to be DTLSRoleClient be an acceptable solution to this?

**Comment by Sean-Der** [comment]
We should totally match the RFC!

If you want to update the value and add a link to the RFC it is an instant merge from me :)
